Gems/GradientSignal/Code/Source/Editor/EditorImageGradientComponentMode.cpp

```diff
@@ -55,45 +147,33 @@ namespace GradientSignal
         Refresh();
 
         m_brushManipulator->Register(AzToolsFramework::g_mainManipulatorManagerId);
+
+        CreateSubModeSelectionCluster();
     }
 
     EditorImageGradientComponentMode::~EditorImageGradientComponentMode()
     {
+        RemoveSubModeSelectionCluster();
+
         AzToolsFramework::PaintBrushNotificationBus::Handler::BusDisconnect();
         m_brushManipulator->Unregister();
 
-        EditorImageGradientRequestBus::Event(GetEntityId(), &EditorImageGradientRequests::SaveImage);
+        EndUndoBatch();
+
+        // It's possible that we're leaving component mode as the result of an "undo" action.
+        // If that's the case, don't prompt the user to save the changes.
+        if (!AzToolsFramework::UndoRedoOperationInProgress())
+        {
+            EditorImageGradientRequestBus::Event(GetEntityId(), &EditorImageGradientRequests::SaveImage);
+        }
 
         ImageGradientModificationBus::Event(GetEntityId(), &ImageGradientModifications::EndImageModification);
         EditorImageGradientRequestBus::Event(GetEntityId(), &EditorImageGradientRequests::EndImageModification);
     }
 
     AZStd::vector<AzToolsFramework::ActionOverride> EditorImageGradientComponentMode::PopulateActionsImpl()
     {
-        return {
-            AzToolsFramework::ActionOverride()
-                .SetUri(PaintbrushIncreaseRadius)
-                .SetKeySequence(QKeySequence{ Qt::Key_BracketRight })
-                .SetTitle(PaintbrushIncreaseRadiusTitle)
-                .SetTip(PaintbrushIncreaseRadiusDesc)
-                .SetEntityComponentIdPair(GetEntityComponentIdPair())
-                .SetCallback(
-                    [this]()
-                    {
-                        AdjustRadius(RadiusAdjustAmount);
-                    }),
-            AzToolsFramework::ActionOverride()
-                .SetUri(PaintbrushDecreaseRadius)
-                .SetKeySequence(QKeySequence{ Qt::Key_BracketLeft })
-                .SetTitle(PaintbrushDecreaseRadiusTitle)
-                .SetTip(PaintbrushDecreaseRadiusDesc)
-                .SetEntityComponentIdPair(GetEntityComponentIdPair())
-                .SetCallback(
-                    [this]()
-                    {
-                        AdjustRadius(-RadiusAdjustAmount);
-                    }),
-        };
+        return m_brushManipulator->PopulateActionsImpl();
```

Comment:
I haven't seen this before and it's an interesting idea to tie manipulator actions to a component mode - one thing to remember is this only exists for component modes and not for the main editor (if we ever want to use the brush manipulator there like we use other manipulators) but it's not a huge problem (this function just wouldn't be called). We might want to update how we handle this in future with the new Action Manager changes @AMZN-daimini is working on too.